### DISCUSSION TITLE
Spectrum OLED does not work

### DISCUSSION THREAD

**Discussion by dukeplay** [rern/rAudio#622] [Fixed Issues]
Hello, I have connected an OLED 128x64 to the I2C. If I activate the Spectrum OLED under System, it works after the reboot. However, the IQuaudio DAC + is then no longer recognized and does not play any sound. After the next reboot, the DAC will work again, but the Spectrum OLED will not. Most OLEDs also have an SSD1306 as a driver, but you can only select the SH1106. Under Moode I already had the OLED running with the SSD1306. Can you change the driver manually in a config and if so where?

**Comment by kan-ibal** [comment]
SSD1306 driver is being shown as Adafruit.

**Reply by dukeplay** [reply]
Many thanks for the answer. I tried the Adafruit, but it only delivers a few pixels. I bought the display with SSD1306, but the SH1106 works better. I still tried, I have to press start and stop several times, then it works. After a short time it goes off again and remains black although the radio station is still on. Sometimes the output is not recognized when booting, sometimes the OLED. Is that due to the I2C bus? Would an OLED with GPIO work better?

**Comment by kan-ibal** [comment]
As far as SSD1306 display is concerned I have to start mpd_oled with parameter -a 3c that is detected using command:
i2cdetect -y 1
and locally built mpd_oled (mpd_oled from Arch crashes or display is distorted, probably due to sync issues).
So I do not userAudio for displaying but mpd audio_output to fifo and
ExecStart=/usr/local/bin/mpd_oled -a 3c -o 3 -b 10 -g 1
in mpd_oled.service override.conf.

**Reply by dukeplay** [reply]
Many thanks for the answer. Would you please describe that to me in more detail? I still have little experience and always need a little guide. In the meantime I've also got an OLED for the SPI to test whether it works with it. At least it is recognized, but it also keeps crashing. Your solution seems to be more stable.

**Reply by kan-ibal** [reply]
Have a look at [https://github.com/antiprism/mpd_oled](https://github.com/antiprism/mpd_oled)

**Comment by dukeplay** [comment]
Fixed with 20211121. Thank you!
